shell/src/main/java/alluxio/cli/ValidateHdfsMount.java

```diff
@@ -0,0 +1,235 @@
+    } catch (IOException e) {
+      return new ValidateUtils.TaskResult(ValidateUtils.State.FAILED, "ufsTests",
+              ValidateUtils.getErrorInfo(e), "");
+    }
+  }
+
+  /**
+   * The entrance.
+   *
+   * @param args command line arguments
+   * */
+  public static void main(String[] args) throws Exception {
+    // TODO(jiacheng): use jccommand?
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd = null;
+    try {
+      cmd = parser.parse(OPTIONS, args, true /* stopAtNonOption */);
+    } catch (ParseException e) {
+      System.exit(1);
+    }
+    args = cmd.getArgs();
+    String ufsPath = args[0];
+
+    InstancedConfiguration conf = InstancedConfiguration.defaults();
+    if (cmd.hasOption(LOCAL_OPTION.getLongOpt())) {
+      // Merge options from the command line option
+      UnderFileSystemConfiguration ufsConf = UnderFileSystemConfiguration.defaults(conf);
+      if (cmd.hasOption(READONLY_OPTION.getLongOpt())) {
+        ufsConf.setReadOnly(true);
+      }
+      if (cmd.hasOption(SHARED_OPTION.getLongOpt())) {
+        ufsConf.setShared(true);
+      }
+      if (cmd.hasOption(OPTION_OPTION.getLongOpt())) {
+        Properties properties = cmd.getOptionProperties(OPTION_OPTION.getLongOpt());
+        ufsConf.merge(properties, Source.MOUNT_OPTION);
+        LOG.debug("Options from cmdline: {}", properties);
+      }
+
+      // Run validateEnv
+      Map<String, String> validateOpts = ImmutableMap.of();
+      ValidateEnv validate = new ValidateEnv(ufsPath, ufsConf);
+      List<ValidateUtils.TaskResult> results = validate.validateUfs(
+              ApplicableUfsType.Type.HDFS, validateOpts);
+
+      // Run runUfsTests
+      if (ufsConf.isReadOnly()) {
+        LOG.debug("Ufs operations are skipped because the path is readonly.");
+        results.add(new ValidateUtils.TaskResult(ValidateUtils.State.SKIPPED,
+                UnderFileSystemContractTest.TASK_NAME,
+                String.format("UFS path %s is readonly, skipped UFS operation tests.", ufsPath),
+                ""));
+      } else {
+        results.add(runUfsTests(ufsPath, new InstancedConfiguration(ufsConf)));
+      }
+
+      // Serialize the results back to the calling node
+      printResults(results);
+
+      System.exit(0);
+    }
+
+    // Cluster mode
+    LOG.info("Invoking the command remotely on the Alluxio cluster.");
+
+    // how many nodes in the cluster
+    Set<String> hosts = ConfigurationUtils.getServerHostnames(conf);
+    ExecutorService executor = Executors.newFixedThreadPool(hosts.size());
+
+    // Invoke validateHdfsMount locally on each host
+    Map<String, CompletableFuture<CommandReturn>> resultFuture = new HashMap<>();
+    for (String host : hosts) {
+      LOG.info("validate hdfs mount on host {}", host);
+
+      // We make the assumption that the Alluxio WORK_DIR is the same
+      String workDir = conf.get(PropertyKey.WORK_DIR);
+      String alluxioBinPath = Paths.get(workDir, "bin/alluxio")
```

Comment:
@madanadit Is the ValidateHdfs required to run on all nodes? @jiacheliu3 I saw the local option and validate hdfs will run by default on local and remote nodes. The problem is this requires SSH and file permissions which is hard to fulfill in real env. I would suggest either remove the remote check (or at least make it optional) or expose validateHdfs rpcs in master/worker. Then we can ask for the server check through rpc channel.

Reply:
In this release (2.3) we target single node (local) validation with no ssh requirement. I agree, remove the remote check.

Reply:
Removed remote check